gitpod-io/gitpod
# [cleanup] remove FF for components service-waiter

## Description


## Related Issue(s)

Fixes [[internal doc](https://www.notion.so/gitpod/service_waiter_skip_components-1f0f683d0a4e438186478b2937a344d8?pvs=4)]

## How to test

Service waiter is still working:
- Open this PR with Gitpod
- Check logs of service-waiter
   - `public-api-waiter`
   - `server-waiter`

✅  server
```
$ kubectl logs dashboard-79f689d5c9-889gt  -c server-waiter
{"component":"server","file":"component.go:46","func":"func2","image":"eu.gcr.io/gitpod-dev-artifact/build/server:commit-f88b2806b53c9a84a53191018c69b096557111b0","labels":"app=gitpod,component=server","level":"info","message":"start to wait component","namespace":"default","serviceContext":{"service":"service-waiter","version":"commit-4986bb82058cec29b23257101404de4e15906539"},"severity":"INFO","time":"2024-06-13T16:34:58.347977881Z","timeout":"5m0s"}
{"@type":"type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent","error":"pod is not ready","file":"component.go:108","func":"waitPodsImage","level":"error","message":"image check failed","serviceContext":{"service":"service-waiter","version":"commit-4986bb82058cec29b23257101404de4e15906539"},"severity":"ERROR","time":"2024-06-13T16:34:58.359789359Z"}
{"@type":"type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent","error":"pod is not ready","file":"component.go:108","func":"waitPodsImage","level":"error","message":"image check failed","serviceContext":{"service":"service-waiter","version":"commit-4986bb82058cec29b23257101404de4e15906539"},"severity":"ERROR","time":"2024-06-13T16:34:59.365377263Z"}
{"file":"export.go:114","func":"Infof","level":"info","message":"ready pods: 1/1","serviceContext":{"service":"service-waiter","version":"commit-4986bb82058cec29b23257101404de4e15906539"},"severity":"INFO","time":"2024-06-13T16:35:00.371217657Z"}
{"file":"export.go:69","func":"Info","level":"info","message":"service is ready","serviceContext":{"service":"service-waiter","version":"commit-4986bb82058cec29b23257101404de4e15906539"},"severity":"INFO","time":"2024-06-13T16:35:00.371278127Z"}
gitpod /workspace/gitpod (hw/ff-2) $ 
```

✅ public-api
```
$ kubectl logs dashboard-79f689d5c9-889gt  -c public-api-server-waiter
{"component":"public-api-server","file":"component.go:46","func":"func2","image":"eu.gcr.io/gitpod-dev-artifact/build/public-api-server:commit-4986bb82058cec29b23257101404de4e15906539","labels":"app=gitpod,component=public-api-server","level":"info","message":"start to wait component","namespace":"default","serviceContext":{"service":"service-waiter","version":"commit-4986bb82058cec29b23257101404de4e15906539"},"severity":"INFO","time":"2024-06-13T16:33:51.685237971Z","timeout":"5m0s"}
{"@type":"type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent","error":"pod is not ready","file":"component.go:108","func":"waitPodsImage","level":"error","message":"image check failed","serviceContext":{"service":"service-waiter","version":"commit-4986bb82058cec29b23257101404de4e15906539"},"severity":"ERROR","time":"2024-06-13T16:33:51.698694559Z"}
{"@type":"type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent","error":"pod is not ready","file":"component.go:108","func":"waitPodsImage","level":"error","message":"image check failed","serviceContext":{"service":"service-waiter","version":"commit-4986bb82058cec29b23257101404de4e15906539"},"severity":"ERROR","time":"2024-06-13T16:34:53.559517356Z"}
{"@type":"type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent","error":"pod is not ready","file":"component.go:108","func":"waitPodsImage","level":"error","message":"image check failed","serviceContext":{"service":"service-waiter","version":"commit-4986bb82058cec29b23257101404de4e15906539"},"severity":"ERROR","time":"2024-06-13T16:34:54.56885079Z"}
{"file":"export.go:114","func":"Infof","level":"info","message":"ready pods: 1/1","serviceContext":{"service":"service-waiter","version":"commit-4986bb82058cec29b23257101404de4e15906539"},"severity":"INFO","time":"2024-06-13T16:34:55.575270304Z"}
{"file":"export.go:69","func":"Info","level":"info","message":"service is ready","serviceContext":{"service":"service-waiter","version":"commit-4986bb82058cec29b23257101404de4e15906539"},"severity":"INFO","time":"2024-06-13T16:34:55.575332384Z"}
```

## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - hw-ff-2</li>
	<li><b>🔗 URL</b> - <a href="https://hw-ff-2.preview.gitpod-dev.com/workspaces" target="_blank">hw-ff-2.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - hw-ff-2-gha.26315</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-hw-ff-2%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>

/hold
